pkg_resources/tests/test_pkg_resources.py

```diff
@@ -237,7 +237,20 @@ def test_version_resolved_from_egg_info(self, env):
         dist = pkg_resources.WorkingSet([env.paths['lib']]).find(req)
         assert dist.version == version
 
-    def test_normalize_path(self, tmpdir):
-        path = str(tmpdir)
-        expected = os.path.normcase(os.path.normpath(os.path.realpath(path)))
-        assert pkg_resources.normalize_path(path) == expected
+    @pytest.mark.parametrize(
+        'suffix, parts',
+        [
+            # Trailing sep should be normalized.
+            ('foo', ('foo')),
+            ('foo/', ('foo')),
+
+            # Casing should be consistent.
+            ('MiXeD/CasE', ('MiXeD', 'CasE')),
+        ],
+    )
+    def test_normalize_path(self, tmpdir, suffix, parts):
+        target = tmpdir.join(*parts)
+        target.mkdir()
+        expected = os.path.normcase(os.path.normpath(os.path.realpath(target)))
```

Comment:
This code just repeats what the primary code does. When I read this test, I can't tell what it's asserting without running it. Better would be to materialize what the expected values are... so you probably want to supply that in the parameters, so that a reader can see what 'MiXeD/CasE' or 'foo/' becomes.

Reply:
@uranusjr If that's the case, I think you have two options:

1. Two separate tests for different platforms
2. Give two non-normalized inputs that should normalize to the same thing and check that `pkg_resources.normalize_path(a) == pkg_resources.normalize_path(b)`

Reply:
You can probably write the second test as a parameterized test, and then have some Windows-specific values that normalize things like `C:\\path\\two\\` and `C:\\path\\two` that are under a `skipif`.

I think all *nix-style paths will be accepted by Windows, but not vice-versa.